enterprise/models.py

```diff
@@ -1961,30 +1961,25 @@ class EnterpriseRoleAssignmentContextMixin(object):
     """
 
     @property
-    def enterprise_customer_uuid(self):
-        """Get the enterprise customer uuid linked to the user."""
-        try:
-            enterprise_user = EnterpriseCustomerUser.objects.get(user_id=self.user.id)
-        except ObjectDoesNotExist:
+    def enterprise_customer_uuids(self):
+        """Get the enterprise customer uuids linked to the user."""
+        enterprise_users = EnterpriseCustomerUser.objects.filter(user_id=self.user.id)
+        if not enterprise_users:
             LOGGER.warning(
                 'User {} has a {} assignment but is not linked to an enterprise!'.format(
                     self.__class__,
                     self.user.id
                 ))
             return None
-        except MultipleObjectsReturned:
-            LOGGER.warning(
-                'User {} is linked to multiple enterprises, which is not yet supported!'.format(self.user.id)
-            )
-            return None
 
-        return str(enterprise_user.enterprise_customer.uuid)
+        return [str(enterprise_user.enterprise_customer.uuid) for enterprise_user in enterprise_users]
 
     def get_context(self):
         """
-        Return the context for this role assignment class.
+        Return the context for this role assignment class. A list is returned in case of user with multiple contexts.
         """
-        return self.enterprise_customer_uuid
+        return str(self.enterprise_customer_uuids[0]) if len(self.enterprise_customer_uuids) == 1 else \
```

Comment:
does calling `self.enterprise_customer_uuids` multiple times here mean that we are recalculating (and thus hitting the db) each time or is that somehow cached? why not always return `self.enterprise_customer_uuids`?

Reply:
We dont want to change the get_context() return type in a non-multiple-context scenario